packages/playback-core/src/index.ts

```diff
@@ -352,6 +351,31 @@ export const loadMedia = (
   }
 };
 
+async function handleNativeError(event: Event) {
+  // If this is a CustomEvent w/ detail return, preventing an infinite loop.
+  if ((event as CustomEvent).detail) return;
+
+  // Stop immediate propagation of the native error event, re-dispatch below!
+  event.stopImmediatePropagation();
+
+  const mediaEl = event.target as HTMLMediaElement;
+  const { message, code } = mediaEl?.error ?? {};
+  const error = new MediaError(message, code);
+
+  if (mediaEl.src) {
+    // Attempt to get the response code from the M3U8 src url.
+    const { status } = await fetch(mediaEl.src as RequestInfo);
```

Comment:
blocking: I think this condition may be too broad? At least in playback-core/mux-video/mux-audio, this may not be an m3u8 and may not be a mux-hosted asset. Also, I think we should be able to check against [the error code](https://developer.mozilla.org/en-US/docs/Web/API/MediaError/code#media_error_code_constants) and only `fetch` for 1 & 2?